### PR TITLE
Added ObjectExtensions tests

### PR DESCRIPTION
**Pull request by goggenb**
Added test cases for ObjectExtensions

### CODE REVIEW

File: Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCastTests.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+﻿namespace MoreDotNet.Tests.Extensions.Common.ObjectExtensions
+{
+    using System;
+    using MoreDotNet.Extensions.Common;
+    using Xunit;
+
+    public class TypeCastTests
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCastTests.cs]: Please rename the test class to AsTests to reflect the method tested.

File: Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCastTests.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        [Fact]
+        public void As_IntAsString_ShouldReturnNull()
+        {
+            object input = 3;
+            var result = input.As<string>();
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void ToDictionary_Null_ShouldThrowArgumentNullException()
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCastTests.cs]: Please move all Test related to ToDictionary to separate class.

File: Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCheckTests.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
+﻿namespace MoreDotNet.Tests.Extensions.Common.ObjectExtensions
+{
+    using System;
+    using MoreDotNet.Extensions.Common;
+    using Xunit;
+
+    public class TypeCheckTests
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCheckTests.cs]: Please rename the test class to IsTests to reflect the method tested.

File: Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCheckTests.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+        [Fact]
+        public void Is_AIsB_ShouldReturnFalse()
+        {
+            var input = new ExampleA();
+            Assert.False(input.Is<ExampleB>());
+        }
+
+        [Fact]
+        public void IsNot_NullItem_ShouldThrowShouldThrowArgumentNullException()
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCheckTests.cs]: Please move the IsNot tests to a separate class.

File: Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCheckTests.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            ExampleA input = null;
+            Assert.Throws<ArgumentNullException>(() => input.IsNot<ExampleB>());
+        }
+
+        [Fact]
+        public void Is_BIsNotA_ShouldReturnTrue()
+        {
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCheckTests.cs]: Please rename to IsNot_BIsNotA_ShouldReturnTrue to reflect the method tested.

File: Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCheckTests.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            var input = new ExampleB();
+            Assert.True(input.IsNot<ExampleA>());
+        }
+
+        [Fact]
+        public void Is_BIsNotB_ShouldReturnFalse()
+        {
```
Comment (Teodor92) [inline Source/MoreDotNet.Test/Extensions/Common/ObjectExtensions/TypeCheckTests.cs]: Same as above.

**Review by Teodor92 — CHANGES_REQUESTED** [review]
Looks good, just a few organizational tweaks needed. Well merge after this :)

**Comment by goggenb** [comment]
Hope i fixed everything with this commit :)

**Review by Teodor92 — APPROVED** [review]

**Comment by Teodor92** [comment]
Merged! Thanks!

**Pull request merged**